src/Nethermind/Nethermind.JsonRpc/Modules/RpcModuleProvider.cs

```diff
@@ -156,21 +211,67 @@ private class ResolvedMethodInfo
             {
                 ModuleType = moduleType;
                 MethodInfo = methodInfo;
-                ExpectedParameters = methodInfo.GetParameters();
+
+                ParameterInfo[] parameters = methodInfo.GetParameters();
+                ExpectedParameter[] expectedParameters = new ExpectedParameter[parameters.Length];
+                for (var i = 0; i < parameters.Length; i++)
+                {
+                    ParameterInfo parameter = parameters[i];
+                    ParameterDetails details = ParameterDetails.None;
+                    if (parameter.ParameterType.IsAssignableTo(typeof(IJsonRpcParam)))
+                    {
+                        details |= ParameterDetails.IsIJsonRpcParam;
+                    }
+                    if (IsNullableParameter(parameter))
+                    {
+                        details |= ParameterDetails.IsNullable;
+                    }
+                    if (parameter.IsOptional)
+                    {
+                        details |= ParameterDetails.IsOptional;
+                    }
+
+                    expectedParameters[i] = new(parameter, details);
+                }
+
+                ExpectedParameters = expectedParameters;
                 ReadOnly = readOnly;
                 Availability = availability;
+                Invoker = MethodInvoker.Create(methodInfo);
```

Comment:
is it faster than? https://learn.microsoft.com/en-us/dotnet/api/system.delegate.createdelegate?view=net-8.0

Could we leverage that?

Reply:
Is faster as it generates and stores the il to invoke (introduced in .NET 8). Delegate is fast if you use the generic version so it's compile time, but will here will be going via `.DynamicInvoke` with object parameters that don't mach the actual parameters (in that they are boxed)